refactor(controller): replace logging statements with zerolog

AxiomaticFixedChimpanzee opened the pull request:
Removes our dependency on `logr` and replaces all log statements with the Zerolog API.

## Pull Request type

<!-- Please try to limit your pull request to one type; submit multiple pull requests if needed. -->

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Fixes: #138

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this does introduce a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

I'm not super happy with how I've introduced the concept of "components" into our logging usage. The desire was to be able to turn debug-logging on/off per "component", so I introduced the ability to attach a component name to a logging instance and enable debug-logging based on a global filter. But since we store our logger in the request context, this implies that the scope of a "component" is dependent on the call path, which is difficult to reason about. If a group of related functions constitute a "component", then `setLogComponent()` needs to be called at the start of all entrypoints (i.e. functions called from outside the component itself), but the component will also "leak" into any downstream called components that don't themselves use `setLogComponent()`. So to keep the concept of log-able components consistent you need to carefully analyze all code paths, and update usages of `setLogComponent()` if anything changes. That doesn't seem very maintainable to me.

Perhaps we would rather have a component be defined in terms of a given file or package, but you can't update the behaviour of a logger stored in the request context based on something static like the package or file it's used from. At least, not without wrapping every call to `log.Ctx()`, which is what we were trying to get away from by using the Zerolog API directly.

In my opinion, "component"-based logging isn't super necessary and it's better to just add more logging context to distinguish log messages, instead of trying to decide beforehand which messages to turn on or off. The whole point of structured logging is to maximize flexibility for filtering after the fact, and modern logging tools make this very easy. If we want to be able to define components based on something static like package or filename, then we could just turn on [file+line based logging context](https://github.com/rs/zerolog?tab=readme-ov-file#add-file-and-line-number-to-log) and use a regex filter. When developing locally, this is as easy as teeing log output to a second terminal, and grepping for some criteria. I often do something like `make start-e2e | tee /dev/pts/2 | grep 'ERR'` to split log output to two terminals, where the second terminal displays all log output, and the first only shows errors. The same could be done for a particular filename or package.

AxiomaticFixedChimpanzee committed:
refactor(controller): replace logging statements with zerolog

portly-halicore-76 reviewed (changes requested):
I hate to bother, but in the previous related PR we've switched to lowercasing log messages. For consistency, I believe we must also change that here. @hikarukin what do you think?

AxiomaticFixedChimpanzee committed:
refactor(controller): transform all log messages to lowercase

Regex used:
`find internal/controller/ -type f | xargs sed -i 's/\(.Msgf\?("\)\([A-Z]\)/\1\L\2/'`

portly-halicore-76 commented on internal/controller/argo_app.go:
```diff
@@ -34,7 +34,8 @@ func (r *PaasNSReconciler) EnsureArgoApp(
 	}
 
-	logger := getLogger(ctx, paasns, "ArgoApp", appName)
+	ctx = setLogComponent(ctx, "ArgoApp")
```
It this printed in log lines as well? If so, than this should be de-capitalised as well?

AxiomaticFixedChimpanzee replied:
It's a field in the logging context, called "component". Some field values are necessarily going to be capitalized, for example our `controller.Kind` context fields, etc. The question is whether we want our component names to be capitalized or not. Personally I feel that we can probably just get rid of these component names entirely, as described in the footnote of my PR description.

AxiomaticFixedChimpanzee replied:
I've gone ahead and changed all component names to lowercase

AxiomaticFixedChimpanzee committed:
refactor(controller): transform all component names to lowercase

portly-halicore-76 reviewed (approved):
LGTM
